airflow/models/dag.py

```diff
@@ -1467,22 +1467,26 @@ def _test_cycle_helper(self, visit_map, task_id):
         """
         from airflow.models.dagbag import DagBag  # Avoid circular imports
 
-        # print('Inspecting %s' % task_id)
-        if visit_map[task_id] == DagBag.CYCLE_DONE:
-            return False
-
-        visit_map[task_id] = DagBag.CYCLE_IN_PROGRESS
-
-        task = self.task_dict[task_id]
-        for descendant_id in task.get_direct_relative_ids():
-            if visit_map[descendant_id] == DagBag.CYCLE_IN_PROGRESS:
-                msg = "Cycle detected in DAG. Faulty task: {0} to {1}".format(
-                    task_id, descendant_id)
+        path, visited = [task_id], []
+
+        while path:
+            current_task_id = path.pop()
+            if visit_map[current_task_id] == DagBag.CYCLE_DONE:
+                return False
+            if visit_map[current_task_id] == DagBag.CYCLE_IN_PROGRESS:
+                msg = "Cycle detected in DAG. Faulty task: {}".format(
+                    current_task_id)
                 raise AirflowDagCycleException(msg)
-            else:
-                self._test_cycle_helper(visit_map, descendant_id)
-
-        visit_map[task_id] = DagBag.CYCLE_DONE
+            visit_map[current_task_id] = DagBag.CYCLE_IN_PROGRESS
+            task = self.task_dict[current_task_id]
+            for descendant_id in task.get_direct_relative_ids():
+                path.append(descendant_id)
```

Comment:
Instead of
```
for descendant_id in task.get_direct_relative_ids():
    path.append(descendant_id)
```
You could simply do `path.extend(task.get_direct_relative_ids())`

Reply:
We might get a few nano seconds out of that :D